ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/B11rSubCommand.java

```diff
@@ -0,0 +1,245 @@
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.hyperledger.besu.evmtool;
+
+import static org.hyperledger.besu.evmtool.B11rSubCommand.COMMAND_ALIAS;
+import static org.hyperledger.besu.evmtool.B11rSubCommand.COMMAND_NAME;
+
+import org.hyperledger.besu.ethereum.core.BlockHeaderBuilder;
+import org.hyperledger.besu.ethereum.mainnet.MainnetBlockHeaderFunctions;
+import org.hyperledger.besu.ethereum.referencetests.BlockchainReferenceTestCaseSpec.ReferenceTestBlockHeader;
+import org.hyperledger.besu.ethereum.rlp.BytesValueRLPOutput;
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.core.JsonParser.Feature;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.tuweni.bytes.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(
+    name = COMMAND_NAME,
+    aliases = {COMMAND_ALIAS},
+    description = "Execute an Ethereum State Test.",
+    mixinStandardHelpOptions = true,
+    versionProvider = VersionProvider.class)
+public class B11rSubCommand implements Runnable {
+  private static final Logger LOG = LoggerFactory.getLogger(B11rSubCommand.class);
+
+  static final String COMMAND_NAME = "block-builder";
+  static final String COMMAND_ALIAS = "b11r";
+  private static final Path stdoutPath = Path.of("stdout");
+  private static final Path stdinPath = Path.of("stdin");
+
+  @Option(
+      names = {"--input.header"},
+      paramLabel = "full path",
+      description = "The block header for the block")
+  private final Path header = stdinPath;
+
+  @Option(
+      names = {"--input.txs"},
+      paramLabel = "full path",
+      description = "The transactions to block")
+  private final Path txs = stdinPath;
+
+  @Option(
+      names = {"--input.ommers"},
+      paramLabel = "full path",
+      description = "The ommers for the block")
+  private final Path ommers = stdinPath;
+
+  @Option(
+      names = {"--seal.clique"},
+      paramLabel = "full path",
+      description = "The clique seal/signature for the block")
+  private final Path sealClique = stdinPath;
+
+  @SuppressWarnings("UnusedVariable") // FIXME do ethash sealing
```

Comment:
Is this for a future PR?

Reply:
t8n tool specifies CLI flags for ethash sealing.  I'm not sure if the retesteth tooling needs it or not, but we don't wire it in.  Much like clique seal we ignore it currently as no meaningful value is passed in by execution-spec-tests, and differential testing shows no diffs relating to extradata in the block (which is where clique seals live).

When I loop back to retesteth support this will get some additional support to pass differential data